### ISSUE TITLE
Exclude Include combination gives wrong results

### ISSUE THREAD

**Issue by nylocx** [python-poetry/poetry#4711]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Arch Linux x64 Kernel 5.14.15
- **Poetry version**: 1.1.11 and 1.2.0a2
- **Python version**: 3.9.7
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [pyproject.toml](https://gist.github.com/nylocx/527a6beacb47ea97d6f7194f7192b5a9)

## Issue

To setup a reproducible environment just run:
```bash
poetry new poetry-cython-demo
cd poetry-cython-demo
mkdir poetry_cython_demo/compiled
touch poetry_cython_demo/cython_code.pyx poetry_cython_demo/compiled/source.c poetry_cython_demo/compiled/source.h
```
and replace the pyproject.toml with the gist version.
Then run:
```bash
poetry build -vvv
```
The expected result is that the sdist package includes the C sources and the .pyx file and the wheel only includes the `__init__.py`.
The actual result is that the wheel includes the .pyx file.

```
❯ poetry build -vvv
…
Building poetry-cython-demo (0.1.0)
  - Building sdist
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/__init__.py
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/cython_code.pyx
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/compiled/source.c
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/compiled/source.h
  - Adding: pyproject.toml
  - Built poetry-cython-demo-0.1.0.tar.gz
  - Building wheel
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/__init__.py
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/cython_code.pyx
…
```

As a side note, my first expectation was that include is working recursively as exclude, so I thought 
`{ path = "poetry_cython_demo/compiled", format = "sdist" }`
would be enough, but than the result is even stranger output where sdist and wheel seam to be switched:
```
❯ poetry build -vvv
…
Building poetry-cython-demo (0.1.0)
  - Building sdist
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/__init__.py
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/cython_code.pyx
  - Adding: pyproject.toml
  - Built poetry-cython-demo-0.1.0.tar.gz
  - Building wheel
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/__init__.py
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/compiled/source.c
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/compiled/source.h
  - Adding: /home/agoertz/projects/poetry-cython-demo/poetry_cython_demo/cython_code.pyx
…
```

**Comment by finswimmer** [comment]
Hello @nylocx,

thanks for reporting. I can confirm this bug and a fix is on the way: https://github.com/python-poetry/poetry-core/pull/228

fin swimmer

**Comment by nylocx** [comment]
Wao, that was a quick one, thanks!

**Comment by nylocx** [comment]
I just checked in it does not work as I expected here is the result of the wheel. It still contains the pyx file. Any idea what is going wrong there? As the test looks like it should check for this.
[poetry_cython_demo-0.1.0-py3-none-any.whl.tar.gz](https://github.com/python-poetry/poetry/files/7767711/poetry_cython_demo-0.1.0-py3-none-any.whl.tar.gz)

**Comment by nylocx** [comment]
@finswimmer I really don't want to annoy you but can you give an approximate ETA when this will land in poetry? I think it was not included in core 1.0.8
